### PR TITLE
[explorer] - [2/2] deprecate DisplayBox 

### PR DESCRIPTION
**Pull request by mamos-mysten**
## Description 

Part 2 of [this PR](https://github.com/MystenLabs/sui/pull/8276) to deprecate / remove the `DisplayBox` component and create a re-usable component for displaying NFT's etc. 

This PR addresses feedback from Part 1, adding a generic Modal / Image componet with image moderation by default.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by mamos-mysten** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8276** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/8276" target="_blank">Graphite</a> 
    * **PR #10864** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/10864" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/MystenLabs/sui/10864?utm_source=stack-comment).

File: apps/explorer/src/ui/ObjectDetails.tsx [hunk 1]
```diff
@@ -38,44 +30,39 @@ const textStyles = cva(['flex min-w-0 flex-col flex-nowrap'], {
...
     id?: string;
-    image: string;
+    image?: string;
     name?: string;
     type: string;
-    nsfw?: boolean;
     variant: 'small' | 'large';
 }
 
 export function ObjectDetails({
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/ObjectDetails.tsx:39]: I feel like this is a little too generic but we can rename in a follow-up

File: apps/explorer/src/ui/modal/Modal.tsx [hunk 2]
```diff
@@ -5,15 +5,47 @@ import { Dialog, Transition } from '@headlessui/react';
...
+
+export function ModalContent({ children }: { children: ReactNode }) {
+    return (
+        <div className="flex flex-col rounded-lg bg-gray-40 p-5">
+            {children}
+        </div>
+    );
+}
+
+export function ModalHeading({ children }: { children: ReactNode }) {
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/modal/Modal.tsx:40]: Seems like these other modal components aren't used? Not sure if that's intentional

File: apps/explorer/src/hooks/useImage.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    const cleanup = () => {
+        if (ref.current) {
+            ref.current.onload = null;
+            ref.current.onerror = null;
+            ref.current = null;
+        }
+    };
+
+    const load = useCallback(() => {
```
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImage.ts:32]: Should this set the status to loading for cases where the `src` changes and it loads a new image?

**Review by Jordan-Mysten — APPROVED** [review]
This diff generally LGTM, let's merge this then iterate in the base branch

**Pull request merged**